### DISCUSSION TITLE
custom button in a single view: +New

### DISCUSSION THREAD

**Discussion by linpi1** [joomengine/Joomla-Component-Builder#800] [Extention Issues]
I created a custom button in an admin view for the single view. It work perfectly when I edit the single view, but not when I click on +New (item). Somehow, in view.html.php & addToolBar(), the custom button is not inserted in "if ($isNew)".
I do make a mistake that the custom button is not inserted in "if ($isNew)" ?

**Comment by Llewellynvdm** [comment]
Yes at the moment this is indeed the case. So since I can't tell what any button will do, I assume all buttons need existing item and so is placed outside the scope of new items.

Not that we can't improve this... but this is how it works now.

All the times so far in the past when I needed it to work on new none existing item, I used the [code replacement method](https://github.com/vdm-io/Joomla-Component-Builder/wiki/TIPS:-Custom-Code).

See this tutorial:
- https://github.com/vdm-io/Joomla-Component-Builder/wiki/031-Automatic-import-of-custom-code-during-compilation-in-JCB

> PS. Thanks for your donation (PRO Membership)

**Reply by linpi1** [reply]
ok thanks, I will try to understand these "script placeholders" and see if I can do it this way. 
Until know, I just add the custom button code by hand, but I have to do it every time after I compile... 
By the way, I'm very enthusiastic about JCB, it is just amazing. I looking forward that JCB will work as well with J!4.

**Reply by linpi1** [reply]
Yes !! it worked, fantastic !

**Comment by TLWebdesign** [comment]
I used this code to insert a button:

**PHP DOCUMENT METHOD:**
`
JToolBarHelper::custom('', 'link button-findtraining', '', JText::_('Find training'), false);
`
I needed the button to trigger a javascript function so i added that attribute via javascript:
`
 document.addEventListener("DOMContentLoaded", function(event) { 
	var elems = document.getElementsByClassName("button-link button-findtraining");
	for(var i = 0; i < elems.length; i++) {
    	elems[i].setAttribute("onclick","findTraining()");
	}
});
`

**Reply by linpi1** [reply]
thank you for this code :-)
